commands/transfer_funds.go

```diff
@@ -144,3 +123,129 @@ func checkTransfer(cmd *commandspb.Transfer) (e Errors) {
 
 	return errs
 }
+
+func validateDispatchStrategy(toAccountType vega.AccountType, dispatchStrategy *vega.DispatchStrategy, errs Errors, prefix string, destinationPrefixErr string) {
+	// check account type is one of the relevant reward accounts
+	if toAccountType != vega.AccountType_ACCOUNT_TYPE_REWARD_LP_RECEIVED_FEES &&
+		toAccountType != vega.AccountType_ACCOUNT_TYPE_REWARD_MAKER_RECEIVED_FEES &&
+		toAccountType != vega.AccountType_ACCOUNT_TYPE_REWARD_MAKER_PAID_FEES &&
+		toAccountType != vega.AccountType_ACCOUNT_TYPE_REWARD_MARKET_PROPOSERS &&
+		toAccountType != vega.AccountType_ACCOUNT_TYPE_REWARD_AVERAGE_POSITION &&
+		toAccountType != vega.AccountType_ACCOUNT_TYPE_REWARD_RELATIVE_RETURN &&
+		toAccountType != vega.AccountType_ACCOUNT_TYPE_REWARD_RETURN_VOLATILITY &&
+		toAccountType != vega.AccountType_ACCOUNT_TYPE_REWARD_VALIDATOR_RANKING {
+		errs.AddForProperty(destinationPrefixErr, ErrIsNotValid)
+	}
+	// check asset for metric is passed unless it's a market proposer reward
+	if len(dispatchStrategy.AssetForMetric) <= 0 && toAccountType != vega.AccountType_ACCOUNT_TYPE_REWARD_MARKET_PROPOSERS && toAccountType != vega.AccountType_ACCOUNT_TYPE_REWARD_VALIDATOR_RANKING {
+		errs.AddForProperty(prefix+".asset_for_metric", ErrUnknownAsset)
```

Comment:
This seems more appropriate:

```suggestion
		errs.AddForProperty(prefix+".asset_for_metric", ErrIsRequired)
```